src/dataframerow/dataframerow.jl

```diff
@@ -278,26 +278,13 @@ Base.merge(a::DataFrameRow, b::DataFrameRow) = merge(NamedTuple(a), NamedTuple(b
 Base.merge(a::DataFrameRow, b::Base.Iterators.Pairs) = merge(NamedTuple(a), b)
 Base.merge(a::DataFrameRow, itr) = merge(NamedTuple(a), itr)
 
-# hash column element
-Base.@propagate_inbounds hash_colel(v::AbstractArray, i, h::UInt = zero(UInt)) =
-    hash(v[i], h)
-Base.@propagate_inbounds function hash_colel(v::AbstractCategoricalArray, i,
-                                             h::UInt = zero(UInt))
-    ref = v.refs[i]
-    if eltype(v) >: Missing && ref == 0
-        hash(missing, h)
-    else
-        hash(CategoricalArrays.index(v.pool)[ref], h)
-    end
-end
-
 # hash of DataFrame rows based on its values
 # so that duplicate rows would have the same hash
 # table columns are passed as a tuple of vectors to ensure type specialization
 rowhash(cols::Tuple{AbstractVector}, r::Int, h::UInt = zero(UInt))::UInt =
-    hash_colel(cols[1], r, h)
+    hash(cols[1][r], h)
```

Comment:
I understand that now it is not needed and proper hashing is ensured by CategoricalArrays.jl?

Reply:
Yes. Actually I'm not even sure it was required before. It's slightly faster, but not that much, and working with `DataFrameRow` is inefficient anyway (we have much more efficient code when hashing a full column, based on hashing each level only once). What do you think?

```julia
julia> df = categorical(DataFrame(rand(["a", "b", "c"], 10, 10)));

julia> r = view(df, 1, :);

# master
julia> @btime hash($r);
  471.533 ns (7 allocations: 352 bytes)

# this PR
julia> @btime hash($r);
  552.241 ns (7 allocations: 352 bytes)

Reply:
It is OK given it is hardly used anyway.